### PR TITLE
Mm 33267/move schemes to schema migrations

### PR DESCRIPTION
**Pull request by nronas**
#### Summary
implements: https://mattermost.atlassian.net/browse/MM-33267

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33267

#### Release Note

```release-note
* Moves schemes gorp driven migrations to schema-based migrations
```

### CODE REVIEW

File: db/migrations/mysql/000013_create_schemes.down.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+DROP TABLE IF EXISTS Schemes;
```
Comment (mgdelacroix) [inline db/migrations/mysql/000013_create_schemes.down.sql:1]: Are we not reversing the alters before the drop anymore? Happy with it, just to acknowledge the change

File: db/migrations/postgres/000013_create_schemes.up.sql [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    deleteat bigint,
+    scope VARCHAR(32),
+    defaultteamadminrole VARCHAR(64),
+    defaultteamuserrole VARCHAR(64),
+    defaultchanneladminrole VARCHAR(64),
+    defaultchanneluserrole VARCHAR(64),
+    UNIQUE(name)
+);
+
+ALTER TABLE schemes ADD COLUMN IF NOT EXISTS defaultteamguestrole VARCHAR(64);
```
Comment (mgdelacroix) [inline db/migrations/postgres/000013_create_schemes.up.sql:17]: I see what you did there ;)

**Review by mgdelacroix — COMMENTED** [review]
Just one open question, otherwise LGTM

File: db/migrations/mysql/000013_create_schemes.down.sql (hunk 1 above)
Comment (nronas) [reply db/migrations/mysql/000013_create_schemes.down.sql:1]: I am 0/5 with this, but i figured what's the chance that a customer will have the main table dropped and not the indexes?

**Pull request merged**